model/src/main/java/org/cloudfoundry/identity/uaa/oauth/jwk/JsonWebKey.java

```diff
@@ -126,32 +127,32 @@ public int hashCode() {
 
     //helper methods
     public String getAlgorithm() {
-        return (String) getKeyProperties().get(ALG);
+        return (String) getKeyProperties().get(HeaderParameterNames.ALGORITHM);
     }
 
     public String getValue() {
-        String result = (String) getKeyProperties().get("value");
+        String result = (String) getKeyProperties().get(PUBLIC_KEY_VALUE);
         if (result == null) {
             if (RSA.equals(getKty())) {
                 result = pemEncodePublicKey(getRsaPublicKey(this));
-                this.json.put("value", result);
+                this.json.put(PUBLIC_KEY_VALUE, result);
             } else if (MAC.equals(getKty())) {
-                result = (String) getKeyProperties().get("k");
-                this.json.put("value", result);
+                result = (String) getKeyProperties().get(JWKParameterNames.OCT_KEY_VALUE);
+                this.json.put(PUBLIC_KEY_VALUE, result);
             }
         }
         return result;
     }
 
     public Set<KeyOperation> getKeyOps() {
-        List<String> result = (List<String>) getKeyProperties().get("key_ops");
+        List<String> result = (List<String>) getKeyProperties().get(JWKParameterNames.KEY_OPS);
         if (result==null) {
             result = Collections.emptyList();
         }
-        return result.stream().map(o -> KeyOperation.valueOf(o)).collect(Collectors.toSet());
+        return result.stream().map(KeyOperation::valueOf).collect(Collectors.toSet());
     }
 
-    public static String pemEncodePublicKey(PublicKey publicKey) {
+    public static String pemEncodePublicKey(Key publicKey) {
```

Comment:
This one feels worse.

Now the method is called `pemEncode` **publicKey**, but the argument is just a `Key`. I don't like this name. What kind of key is it? The key to my bike lock?

Reply:
Sonar suggested this to me